### PR TITLE
docs: updated documentation

### PR DESCRIPTION
**Pull request by dharmez**
Updated README.md and DATAFLOW.md

### CODE REVIEW

File: assets/deployment-architecture.png
Comment (eeaton) [inline assets/deployment-architecture.png:1]: I have a number of small comments with the diagram, perhaps it would be easier to review and edit in slides directly instead of Github?

 - unusual abbreviation "bank stmts", write it out as "bank statements"
 -  if we're using bullets to separate items, don't need "," or "&" between items
 - replace "employee (IT) OR Cron Job" with "Uploader persona"
 - replace "empoyee biz ops" with "Reader persona"
 - "Excel, Outlook Processor" isn't called this in the job names, which job does it refer to? 
 - bubble "11" should be between Web UI and the Reader, not between Web UI and Search App
 - text for number 5 references "AlloyDB, BigQuery and Firestore" but the diagram avoids BigQuery, is this deliberate?
 - numbering is off, the diagram has 12 but text has 10. I think the gap is 9,10,12 on diagram don't exist in text.

File: DATAFLOW.md [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
 1. **Upload documents and trigger document processing workflow**
-   An employee (IT) manually uploads various document types (Excel, PDF, Word, Text, Outlook, .xlsx, .pdf, .docx) or a scheduled Cron Job triggers the upload. The uploaded files are stored in a designated "input" bucket within Google Cloud Storage.
-
-   The upload to the "input" bucket triggers a Document Processing Workflow managed by Cloud Composer.
-
+   An employee (IT) manually uploads various document types (Excel, PDF, Word, Text, Outlook) or a scheduled Cron Job triggers the upload. The uploaded files are stored in a designated "input" bucket within Google Cloud Storage.
+   
+   The upload to the "input" bucket triggers a Document Processing Workflow managed by Cloud Composer.  
 1. **Validate the documents:**
    Documents that fail validation are redirected to a separate "reject" bucket within Google Cloud Storage for further review and handling.
```
Comment (eeaton) [inline DATAFLOW.md:9]: I don't think we ever explained the criteria for how documents are rejected. Is it unsupported file types?

File: DATAFLOW.md [hunk 2]
```diff
@@ -1,28 +1,27 @@
...
 1. **Initiate Processing Jobs:**
    The workflow initiates specific processing jobs based on the document type:
-   `PDF Classifier (Cloud Run -> Doc AI Custom Classifier):` Identify PDF documents that contain forms.
-   `Forms Processor (Cloud Run -> Doc AI Form Parser):` Processes form documents(PDF).
-   `Excel, Outlook Processor (Cloud Run):` Processes Excel and Outlook files.
```
Comment (eeaton) [inline DATAFLOW.md:14]: Names of jobs here don't match the names of Jobs that we create, so it's hard to understand which is referring to which

File: DATAFLOW.md [hunk 3]
```diff
@@ -1,28 +1,27 @@
...
-
+   An employee (IT) manually uploads various document types (Excel, PDF, Word, Text, Outlook) or a scheduled Cron Job triggers the upload. The uploaded files are stored in a designated "input" bucket within Google Cloud Storage.
+   
+   The upload to the "input" bucket triggers a Document Processing Workflow managed by Cloud Composer.  
 1. **Validate the documents:**
    Documents that fail validation are redirected to a separate "reject" bucket within Google Cloud Storage for further review and handling.
 1. **Initiate Processing Jobs:**
    The workflow initiates specific processing jobs based on the document type:
-   `PDF Classifier (Cloud Run -> Doc AI Custom Classifier):` Identify PDF documents that contain forms.
-   `Forms Processor (Cloud Run -> Doc AI Form Parser):` Processes form documents(PDF).
```
Comment (eeaton) [inline DATAFLOW.md:13]: We use "process" and "parser" interchangeably, but don't explain what either means in this context. Needs more explanation of what it does, maybe something like "processes form documents by reading the document and extracting fields that will be later used for RAG" (I'll defer to you if there's a better description)

File: DATAFLOW.md [hunk 4]
```diff
@@ -1,28 +1,27 @@
...
-   The extracted document metadata is stored in a BigQuery table.
+   - `Document Classifier (Cloud Run -> Doc AI Custom Classifier):`  Identify PDF document type (Any form, bank statements, invoices, pay slip, lending doc and utility bills)  
+   - `Specialized Parser (Cloud Run -> Doc AI Form and Specialied Parser):` Processes forms and documents (Any form, bank statements, invoices, pay slip, lending doc and utility bills)  
+   - `Excel, Outlook Processor (Cloud Run):` Processes Excel and Outlook files.  
 1. **Utilize Document AI Batch API:**
-   The Document Processing Workflow leverages the Document AI Batch API for asynchronous processing of large document volumes.
-1. **Create and store document metadata for forms in PDF documents:**
-   The extracted document metadata is stored in a BigQuery table.
+   The Specialized Parser Job leverages Document AI Batch API for asynchronous processing of large document volumes.
+1. **Extract and store data elements:**
```
Comment (eeaton) [inline DATAFLOW.md:17]: It's hard to follow how steps 5, 6, and 7 are different. They're saying mostly the same thing but highlight different data products, but it's not clear to the reader why the multiple sources are required or what outcomes are enabled by the multiple sources

File: DATAFLOW.md [hunk 5]
```diff
@@ -1,28 +1,27 @@
...
+   The Specialized Parser Job uses Document AI Batch API to extract data elements from the documents and stores the extracted data elements in AlloyDB, BigQuery and Firestore.  
+1. **Create and store document metadata in BigQuery for Search and Summarization:**
+   Document processsing jobs creates and stores document metadata in a BigQuery table. Document metadata is required by Agent Builder for indexing the documents. Document metadata contains the uri to the documents stored in the "process" bucket in Google Cloud Storage.
+1. **Create and store processed documents in Cloud Storage Bucket for Search and Summarization:**
+   The processed documents and document metadata files are also stored in the "process" bucket in Google Cloud Storage for search and summarization.
 1. **Data Serving:**
    The processed and stored content from the "process" bucket is imported and indexed in Agent Builder’s BigQuery Data Store.
```
Comment (eeaton) [inline DATAFLOW.md:24]: "GCS data is imported... to Agent Builder's **BigQuery** Data Store" --> is this correct?
I thought the data store could be either GCS-based or BQ-based depending on the input of source files.  The diagram shows both but the text makes it sound like GCS inputs are converted to a BQ-based datastore. 

To make clearer, how about something that generalizes across both sources like "Data and metadata from the processed documents is imported into an Agent Builder Data Store"

File: README.md [hunk 6]
```diff
@@ -52,9 +52,25 @@ To deploy this solution, perform the follow steps:
     - Service Account Admin
     - Service Usage Admin
 
-1.  To deploy this repository using an online terminal with software and authentication preconfigured, use [Cloud Shell](https://shell.cloud.google.com/?show=ide%2Cterminal).
+1.  Create, train and deploy a custom document classifier
 
-    Alternatively, to deploy this repository using a local terminal:
+    To classify documents, you must [create a custom document classifier in the Google Cloud console](https://cloud.google.com/document-ai/docs/custom-classifier).
```
Comment (eeaton) [inline README.md:57]: I disagree with moving the custom classifier steps to here. Most of this section is about how to deploy the Terraform resources self-contained in this repo, so it's out of place to require elaborate steps that aren't part of the repo.

(Same argument for the step about the domain)

I suggest we make this subsectionection only for deploying the Terraform. How about restructuring like...
1. Deployment Guide
    1. Deploy the terraform
    2. Configure the custom classifier (this content)
    3. Enable the web application (steps about the domain name ownership, and additional steps about DNS and OAuth config that currently aren't documented)

File: DATAFLOW.md (hunk 1 above)
Comment (eyalbenivri) [reply DATAFLOW.md:7]: I'm not sure if we explained it or not, but if not, we definitely should. Unsupported file types is one, but we also reject files that are too big, or files that we detected as duplicates.

File: DATAFLOW.md (hunk 5 above)
Comment (eyalbenivri) [reply DATAFLOW.md:22]: I'll just say that documents are kept in GCS, but data extracted from the document is kept in BQ and AlloyDB, plus metadata in BQ, plus ingested in Agent builder, which is the secret sauce (vectorized version of the document).

So combination of all.

File: README.md [hunk 7]
```diff
@@ -105,24 +105,20 @@ To deploy the Infrastructure-as-Code (IaC) resources needed for this solution, p
...
-
-    - (Optional) By default, the Terraform script creates a new VPC network in the same project as other resources. You can use an existing VPC network instead by configuring the following optional terraform variables.
-
-      If using an existing VPC, you should validate that your existing vpc has firewall policies and DNS zones that enable the traffic pathways defined in [vpc.tf](enterprise-knowledge-solution/components/common-infra/terraform/vpc.tf), and grant Compute Network User on your shared VPC to the deployer service account.
-
-      ```sh
-      create_vpc_network = false # default is true
-      vpc_name = # The name of your existing vpc, (e.g., "myvpc")
-      ```
+      | Terraform variables         | Description                                                                                          |
```
Comment (eeaton) [inline README.md:108]: In my previous commit I changed this section back the to the shell code format, and it looks like you reverted it back again to the table. Why the table?

My view is that the lines of shell are easier to copy paste with a few clicks to modify the values. The table format is harder to use because they have to copy paste each line individual or reformat it somewhere else before they can put it in a shell.

File: README.md [hunk 8]
```diff
@@ -174,10 +170,9 @@ Some workflows depend on a [custom classifier](https://cloud.google.com/document
 1. After deployment is complete, click the three dot menu again and choose "set as default".
 
 1. After all steps to train and deploy the customer classifier are complete, add the following variable to your `terraform.tfvars` and run `terraform apply` again.
-
```
Comment (eeaton) [inline README.md:158]: same as other comments about shell vs table for inputting shell variables

File: README.md [hunk 9]
```diff
@@ -284,9 +279,9 @@ Once the workflow completes successfully, all documents will be imported into th
    - On the console page, you'll find an input bar. Enter your questions or queries related to the documents you've uploaded.
    - The app will provide summarized answers based on the content of your documents, along with references to the specific source documents.
 
-### Search and Explore from EKS Web UI
+### Search and Explore from EKS Web UI (Skip this if you have not configured the access to the Web UI Search Application)
```
Comment (eeaton) [inline README.md:282]: Nit: it's odd to give full sentence directions in a section header, how about the following...

Instead of 
```
### Search and Explore from EKS Web UI (Skip this if you have not configured the access to the Web UI Search Application)
```

I suggest:
```
### Search and Explore from EKS Web UI
_Note: This section only applies if you completed the "configure access to the Web UI search application" step under the deployment guide. 
```

File: README.md [hunk 10]
```diff
@@ -82,85 +82,118 @@ To deploy this solution, perform the follow steps:
...
+      | project_id                  | Your Google Cloud project ID.                                                                        |
+      | region                      | The desired region for deploying single-region resources (e.g., "us-central1", "europe-west1").      |
+      | vertex_ai_data_store_region | The multiregion for your Agent Builder Data Store, the possible values are ("global", "us", or "eu").|
+      | docai_location              | Sets the location for Document AI                                                                    |
+      | webui_domains               | Your domain name for Web UI access (e.g., ["webui.example.com"])                                     |
+      | iap_access_domains          | List of domains granted for IAP access to the Web UI (e.g., ["domain:example.com"])                  |
+
+1.  (Optional) By default, the Terraform script creates a new VPC network in the same project as other resources. You can use an existing VPC network instead by configuring the following optional terraform variables.
+      | Terraform variables                 | Description                                    |
```
Comment (eeaton) [inline README.md:118]: Line 118 introduced trailing whitespace that causes issues with Lint.

You can fix this and similar issues by running:
```
export LINTER_CONTAINER_FIX_MODE=true
scripts/lint.sh
```

Or, I configure my IDE with a devcontainer so that it will automatically configure the built-in linters to catch and fix this on every save.

**Comment by eeaton** [comment]
I've made a few further comments, they don't need to be blocking so I'll give an approval now but make sure to resolve those and fix lint issues before merging.

**Review by eeaton — APPROVED** [review]
I've made a few further comments, they don't need to be blocking so I'll give an approval now but make sure to resolve those and fix lint issues before merging.

File: README.md (hunk 7 above)
Comment (dharmez) [reply README.md:108]: The users should clearly understand these setting and so I think we should keep this in a table form so that when they do it one at a time, they think rather than blank copy-paste.

File: DATAFLOW.md (hunk 1 above)
Comment (dharmez) [reply DATAFLOW.md]: We are not rejecting the duplicates at the moment.

File: assets/deployment-architecture.png
Comment (dharmez) [reply assets/deployment-architecture.png:1]: I fixed the numbering in the DATAFLOW.md

File: README.md (hunk 9 above)
Comment (dharmez) [reply README.md:282]: Yes, I will make this change.

File: DATAFLOW.md [hunk 11]
```diff
@@ -1,28 +1,42 @@
-# Enterprise Knowledge Solution (EKS) Dataflow
+# Enterprise Knowledge Solution Dataflow
 
-1. **Upload documents and trigger document processing workflow**
-   An employee (IT) manually uploads various document types (Excel, PDF, Word, Text, Outlook, .xlsx, .pdf, .docx) or a scheduled Cron Job triggers the upload. The uploaded files are stored in a designated "input" bucket within Google Cloud Storage.
-
-   The upload to the "input" bucket triggers a Document Processing Workflow managed by Cloud Composer.
+1. **Upload documents and trigger document processing workflow:**
+   The [uploader persona](docs/access_control.md) uploads documents (Excel, PDF, Word, Text, Outlook, Zip) to a designated "input" bucket in Google Cloud Storage. This persona might be an employee responsible for manually curating and uploading the content, or it might be an automated cron job that transfers documents from a source system.
```
[the remark this answers is not in the document]
Comment (dharmez) [reply DATAFLOW.md:4]: Ok, I will update it.

File: README.md (hunk 10 above)
Comment (dharmez) [reply README.md:118]: Ok, thanks!

**Review by eeaton — APPROVED** [review]

**Review by dharmez — COMMENTED** [review]
LGTM

**Review by eeaton — APPROVED** [review]
